### PR TITLE
pressing delete while in vector mask mode should sometimes delete the…

### PR DESCRIPTION
**Pull request by pineapplespatula**
… vector mask

fix for issue #2860

### CODE REVIEW

File: src/js/actions/layers.js [hunk 1]
```diff
@@ -2225,6 +2225,54 @@ define(function (require, exports) {
...
+                        event = events.document.history.optimistic.REMOVE_VECTOR_MASK_FROM_LAYER;
+
+                    return this.dispatchAsync(event, payload);
+                });
+        } else {
+            return Promise.resolve();
+        }
+    };
+    deleteVectorMask.reads = [locks.JS_TOOL];
+    deleteVectorMask.writes = [locks.PS_DOC];
```
Comment (mcilroyc) [inline src/js/actions/layers.js]: I think this writes JS_DOC as well

**Comment by mcilroyc** [comment]
Aside from the lock issue, I have two functional issues for you:
1) I can't delete the mask when in SS tool?
2) When I delete the mask in elip tool, it deletes the whole layer!

http://cl.ly/132R0L0N3U2N
part way through when I jump back in to mask mode while in SS tool, you can't see on the video but I'm pressing the delete key to no avail.  Then later you'll see I switch to E tool and press delete.

**Comment by pineapplespatula** [comment]
good catch. @mcilroyc  back to you

File: src/js/tools/superselect/vector.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+    /*
```
Comment (mcilroyc) [inline src/js/tools/superselect/vector.js]: wut

**Pull request merged**